### ISSUE TITLE
hvd.size() always gets 1 with unreleased version

### ISSUE THREAD

**Issue by PANXiao1994** [horovod/horovod#1646]
**Environment:**
1. Framework: TensorFlow
2. Framework version: 1.15.0
3. Horovod version: unreleased version by using `HOROVOD_NCCL_HOME=/usr HOROVOD_GPU_BROADCAST=NCCL HOROVOD_WITH_TENSORFLOW=1 HOROVOD_GPU_ALLREDUCE=NCCL pip3 install git+https://github.com/horovod/horovod.git`
4. MPI version: Open MPI: 2.0.2
5. CUDA version: CUDA Version 10.0.130
6. NCCL version: libnccl-dev=2.4.8-1
7. Python version: python 3.5
8. OS and version: Debian GNU/Linux,   9 (stretch)
9. GCC version: gcc version 6.3.0 20170516 (Debian 6.3.0-18+deb9u1)

**Bug report:**
Please describe errorneous behavior you're observing and steps to reproduce it.

In order to use the latest feature [`backward_passes_per_step`](https://github.com/horovod/horovod/blob/master/horovod/tensorflow/__init__.py#L443-L446) on Tensorflow, I am re-installing from github by this command:

``` 
HOROVOD_NCCL_HOME=/usr HOROVOD_GPU_BROADCAST=NCCL HOROVOD_WITH_TENSORFLOW=1 HOROVOD_GPU_ALLREDUCE=NCCL pip3 install git+https://github.com/horovod/horovod.git
```
However, after re-installing, I found that `hvd.size()` always gives 1, even when I have multiple GPUs.

I test `hvd.size()` by this script:
```
import horovod.tensorflow as hvd
# Horovod: initialize Horovod.
hvd.init()
print(hvd.size())
```
and run:
```
mpirun --allow-run-as-root -np 4     -bind-to none -map-by slot     -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH -x PATH     -mca pml ob1 -mca btl ^openib     python3 run.py
```

The log is as follows:
![image](https://user-images.githubusercontent.com/24204710/72055569-054d6500-3306-11ea-8d87-e3d6757b3fcf.png)

P.s. The above command prints `hvd.size()=4` correctly with horovod released version 0.18.2.

OR: Is there any other way to implement gradient accumulation with TensorFlow?

**Comment by tgaddair** [comment]
Hey @PANXiao1994, I wasn't able to reproduce this in my environment.  Can you try modifying your `run.py` script with the following:

```
 from mpi4py import MPI
 comm = MPI.COMM_WORLD
 print(comm.Get_size())
```

You can install `mpi4py` with `pip install mpi4py`.  I want to rule out the possibility that the issue is coming from the MPI side.

**Comment by PANXiao1994** [comment]
> from mpi4py import MPI
>  comm = MPI.COMM_WORLD
>  print(comm.Get_size())

Thanks, it perfectly solved my problem!

**Comment by tgaddair** [comment]
Hey @PANXiao1994, to clarify, `comm.Get_size()` returned 4 for `hvd.size()` still returns 1?  If so, it sounds like there's a Horovod issue.

**Comment by zhangjunnana** [comment]
Hey @tgaddair I had the same problem.  The hvd.size() was always 1. I had tried comm.Get_size(), it still returned 1. My machine has 3 GPUs(RTX 2080Ti). 

Environment:

- Framework: TensorFlow 1.14.0
- Horovod version: 0.19
- MPI version: Open MPI: 4.03
- CUDA version: CUDA Version 10.1
- NCCL version: libnccl-dev=2.5.6
- Python version: python 2.7

**Comment by tgaddair** [comment]
Hey @zhangjunnana, if `comm.Get_size()` is also returning 1, then there is a problem with your MPI configuration.  

Are you using `horovodrun`? If so, you may want to consider running with `horovodrun --gloo ...` to avoid having to mess with MPI.

If you want/need to stick with MPI, then usually this is caused by having multiple versions of MPI in your system.  See: https://stackoverflow.com/questions/4039608/why-do-all-my-open-mpi-processes-have-rank-0

**Comment by zhangjunnana** [comment]
@tgaddair  Thanks !!!   `horovodrun --gloo ...` works.

**Comment by talkhaldi** [comment]
In my case, hvd.size() always = 1, comm.Get_size() = 2. I'm not sure how to debug this...

I use this command:
`mpirun -np 2 -map-by ppr:2:node -mca btl_tcp_if_include enp129s0f0,lo python my_script.py`

Thanks in advance!

**Comment by talkhaldi** [comment]
Okay, I'm not sure what exactly happened, but I think I got it fixed by using HOROVOD_WITH_MPI=1 when installing horovod. On another cluster that I've been using, the instructions of installing horovod didn't include HOROVOD_WITH_MPI and it still worked with MPI. I'm not sure why now I needed to specify it explicitly.

Anyways, sorry for the disruption!
